src/vs/workbench/services/history/browser/historyService.ts

```diff
@@ -162,7 +162,7 @@ export class HistoryService extends Disposable implements IHistoryService {
 		// Listen to selection changes if the editor pane
 		// is having a selection concept.
 		if (isEditorPaneWithSelection(activeEditorPane)) {
-			this.activeEditorListeners.add(activeEditorPane.onDidChangeSelection(e => this.handleActiveEditorSelectionChangeEvent(activeEditorGroup, activeEditorPane, e)));
+			this.activeEditorListeners.add(Event.accumulate(activeEditorPane.onDidChangeSelection, 0, this._store)(e => this.handleActiveEditorSelectionChangeEvent(activeEditorGroup, activeEditorPane, e)));
```

Comment:
I think this is introducing a leak right? You register on `this._score` but `activeEditorListeners` is the one that is being cleaned up.